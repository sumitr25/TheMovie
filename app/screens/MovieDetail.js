import React, { useEffect, useState } from 'react';
import { ActivityIndicator, Image, StyleSheet, Text, View } from 'react-native';
import colors from '../res/colors';
import fonts from '../res/fonts';
import { getMovieDetail } from '../services';
import { fullUrl } from '../services/utils';

const MovieDetail = ({ route }) => {
  const id = route.params;
  const [movieDetail, setMovieDetail] = useState({});
  const [isLoaded, setIsLoaded] = useState(true);

  useEffect(() => {
    getMovieDetail(id.itemId)
      .then(response => {
        setIsLoaded(false);
        setMovieDetail(response.data);
      })
      .catch(error => {
        setIsLoaded(false);
        console.log(error);
      });
  }, []);

  return (
    <View style={styles.container}>
      {!isLoaded ? (
        <View>
          <Image
            style={styles.banner}
            source={{
              uri: fullUrl(movieDetail.poster_path),
            }}
          />
          <Text style={styles.title}>{movieDetail.title}</Text>
          <Text style={styles.description}>{movieDetail.overview}</Text>
        </View>
      ) : (
        <ActivityIndicator
          size={'large'}
          color={colors.headerColor}
          style={styles.activityIndicator}
        />
      )}
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: 'white',
    paddingHorizontal: fonts.large,
    paddingVertical: fonts.small,
    flex: 1,
  },
  banner: {
    height: fonts.bannerHeight,
  },
  title: {
    color: colors.black,
    fontWeight: fonts.bold,
    fontSize: fonts.large,
    paddingVertical: fonts.small,
  },
  description: {
    color: colors.black,
  },
  activityIndicator: {
    flex: 1,
  },
});

export default MovieDetail;
