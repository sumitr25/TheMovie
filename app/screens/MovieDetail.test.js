import React from 'react';
import {shallow} from 'enzyme';
import MovieDetail from './MovieDetail';

describe('MovieDetail', () => {
    describe('Rendering', () => {
        it('should match to snapshot', () => {
            const component = shallow(<MovieDetail />)
            expect(component).toMatchSnapshot()
        });
    });
});