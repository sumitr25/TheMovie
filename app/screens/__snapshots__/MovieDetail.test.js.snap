// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetail Rendering should match to snapshot 1`] = `ShallowWrapper {}`;
